docs/dev/reference/hooks/validateFormField.md

```diff
@@ -55,7 +55,7 @@ class ValidateFormFieldListener implements ServiceAnnotationInterface
      */
     public function onValidateFormField(Widget $widget, string $formId, array $formData, Form $form): Widget
     {
-        if ('myform' === $formId && $widget instanceof \Contao\FormText) {
+        if ('auto_form_42' === $formId && $widget instanceof \Contao\FormTextField && 'mywidget' === $widget->name) {
```

Comment:
Die Auto-ID wäre nicht zulässig, wenn man einen Hook verwenden und über die Form-ID das Formular identifizieren möchte. Man muss natürlich auch eine Form-ID im Backend beim jeweiligen Formular vergeben.

Reply:
warum ist die Auto-ID nicht "zulässig"? ... kann ich auch wieder zurück nehmen...
wichtig ist die Korrektur bei instanceof

Reply:
> warum ist die Auto-ID nicht "zulässig"? ... kann ich auch wieder zurück nehmen...

Weil das halt keinen Sinn macht, du darfst dich ja in deinem Hook Code nicht auf etwas verlassen, das von Contao automatisch generiert wird. Du musst dein Formular eindeutig identifizieren können, egal was Contao macht.

> ich glaube nicht, dass das "allgemein bekannt" ist - frage mich, wo man den Hinweis unterbringen könnte; eine Zusammenfassung für alle Hooks, die das Formular umfassen gibt ea ja nicht

Naja, jemand der einen Hook implementiert muss sich ja selbst fragen "wie kann ich sicherstellen, dass mein Hook genau für das Formular, oder die Formulare, ausgeführt wird, die ich haben möchte". Und da sollte es ja wohl klar sein, dass man sich nicht auf etwas generisches verlassen kann, was sich jederzeit ändern kann ;).

> wichtig ist die Korrektur bei instanceof

Jap, das passt.